go/vt/sqlparser/ast.go

```diff
@@ -4845,6 +4846,117 @@ func (node *Order) walkSubtree(visit Visit) error {
 	)
 }
 
+type FrameUnit int
+
+const (
+	// RangeUnit matches by value comparison (e.g. 100 units cheaper)
+	RangeUnit FrameUnit = iota
+	// RowsUnit matches by row position offset (e.g. 1 row before)
+	RowsUnit
+)
+
+// Frame represents a window Frame clause.
+type Frame struct {
+	Unit   FrameUnit
+	Extent *FrameExtent
+}
+
+// Format formats the node.
+func (node *Frame) Format(buf *TrackedBuffer) {
```

Comment:
I don't actually know how these tests pass

Reply:
The reason they pass is because you have been fooled: in these tests, select expressions echo their textual input *unless* you tell the test case `serializeSelectExprs: true`

When you add that to one of these cases, you can see they don't actually work, no range info is printed:

expected: "select name, dense_rank() over (partition by x order by y RANGE BETWEEN 1 PRECEDING AND 1 FOLLOWING) from t"
        	            	actual  : "select name, dense_rank() over (partition by x order by y asc) from t"

From a brief look, it seems most of the OVER tests have this issue

Reply:
the opposite is now default, need to specify `useSelectExpressionLiteral` to sidestep parsing